core/src/main/java/org/infinispan/notifications/cachelistener/event/impl/EventImpl.java

```diff
@@ -175,7 +171,7 @@ public V getOldValue() {
    }
 
    public void setValue(V value) {
-      this.value = value;
+      this.value = MarshalledValue.unwrap(value);
```

Comment:
I like the lazy approach here, because the value might not be accessed by the listener. Or there may even be no listeners attached...